drivers/usb/device/usb_dc_stm32.c

```diff
@@ -208,16 +208,67 @@ void HAL_PCD_SOFCallback(PCD_HandleTypeDef *hpcd)
 }
 #endif
 
-static int usb_dc_stm32_clock_enable(void)
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32u5_otghs_phy)
+
+static const struct stm32_pclken phy_pclken[] = STM32_DT_CLOCKS(DT_INST_PHANDLE(0, phys));
+
+static int usb_dc_stm32u5_phy_clock_select(const struct device *const clk)
```

Comment:
@erwango Why is it not in the clock control driver, and why do we have to deal with all this unrelated driver code here? How will the changes be synchronized with the new driver?

Reply:
I understand your concern, but this configuration is purely out of clock control handling and depends on USB.
Note that even isn't a "RCC" block configuration, but a "SYSCFG" (System configuration controller) one.
Clock controller driver doesn't have information about the USB context, it "just" enables/disables clocks and has the knowledge about the related rate. Configuring which clock should be used it out if its scope and depend on consuming IP.

> How will the changes be synchronized with the new driver?
For sure, we're planning to port this to the new driver (see my above comment), but I didn't want to make this blocking for @mniestroj.

Reply:
@jfischer-no Please comment.

Reply:
FYI, just created https://github.com/zephyrproject-rtos/zephyr/issues/82745 for tracking

Reply:
@jfischer-no PTAL. Your legitimate concern about backport is I think addressed. For the other point about clock-control architecture, while I disagree, can be discussed later on but let's avoid scope creep in current PR.

Reply:
> We also have drivers/usb/common, you can put code there that is used by the both drivers

Interesting thanks. We'll surely consider it. Adding to comment to `drivers/usb/common`